### ISSUE TITLE
Exclude custom resources by path

### ISSUE THREAD

**Issue by aleske** [asciidoctor/asciidoctor-maven-plugin#217]
Hi,

I'm generating documentation in multiple projects stored in SVN :-/
Each doc include the same header and footer file. These files are in a specific svn project.
In order to make this configuration work, I use svn:externals property.
Problem is .svn directory is created in the sourceDirectory.

Simple example
```
/doc/source/common/.svn <- here is the problem
/doc/source/common/header.adoc
/doc/source/myDoc.adoc
```

When copying resources, the FileFilter used only excludes asciidoc files. I don't want (and Jenkins is not allowed to do it anyway) svn files to be copied into the outputDirectory.

It would be a great idea to add a customizable FileFilter to permit exclusion like maven-site-plugin provides via 
```xml
<moduleExcludes>
    <asciidoc>**/_*.adoc</asciidoc>
</moduleExcludes>
```

Seems legit ?

**Comment by abelsromero** [comment]
I feel this keeps biting from time to time...this has been in my head for some time.
Instead of just excluding, the idea (taken from the gradle plugin) is to add a <resources> section to filter which ones you want to copy. The resources section will work like in the maven-resources-plugin, so it should be familiar to most maven users and it allows to both define include patterns and exclude. What do you think?

You can find some preview implementation here (ignore the sources section, this will be for v1.6.+):
https://github.com/asciidoctor/asciidoctor-maven-plugin/pull/193

Let's use this conversation to gather all options and define the final solution. Do you think this can fully solve your case? any other scenario you can think off?

**Comment by abelsromero** [comment]
PS: Cool image, brings back some childhood memories

**Comment by abelsromero** [comment]
Hi @aleske, last PR should fix this issue. Before closing I'd like to check with you. 
A new element has been added to allow resource filtering following the same patterns as the maven-resources-plugin.

I have published a preview to test, the details are here:
https://bintray.com/abelsromero/maven/asciidoctor-maven-plugin/1.5.4-preview.2/


You can do something like this now:
```
<resources>
    <resource>
        <directory>${target_path}</directory> <!-- or just use . o filter across all files-->
        <exclude>**/**</exclude> <!-- does not copy anything -->
    </resource>
</resources>

**Comment by aleske** [comment]
Oh ! I'm sorry for not having answered your previous message. I was busy as hell.

Anyway it seems perfect for me. I'm able to exclude svn files/folder and even the damned AsciidocFX generated folders (sometimes people commit somes).

Thanks !

**Comment by abelsromero** [comment]
> Oh ! I'm sorry for not having answered your previous message. I was busy as hell ... then I forgot.

Do not worry at all :smile: . Happy it fixes your case, in that case I'll close the issue.

**Comment by abelsromero** [comment]
Closed with:
https://github.com/asciidoctor/asciidoctor-maven-plugin/pull/225
https://github.com/asciidoctor/asciidoctor-maven-plugin/pull/226

To be released with 1.5.4
